### PR TITLE
Add simple Git configure action

### PR DESCRIPTION
**Pull request by jpalomaki**
Configures `git` and `gh` CLIs for use in later steps. Sample usage:
```
- name: Configure git
  uses: pafcloud/terraform-actions/github/configure-git@TODO
  with:
    user-name: somebot
    user-email: somebot@<tld>
    gpg-key-id: ABC1234567890DEF
    gpg-key-pem: ${{ secrets.GPG_KEY }}
    ssh-key-pem: ${{ secrets.SSH_KEY }}
    github-token: ${{ secrets.GITHUB_TOKEN }}
```
Includes git GPG signing key setup, so we can issue signed commits with a bot user.

Operator only needs to pass necessary inputs (as not all may be needed in all use cases).

### CODE REVIEW

File: github/configure-git/action.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+runs:
+  using: composite
+  steps:
+    - run: |
+        git config user.name "$USER_NAME"
+        git config user.email "$USER_EMAIL"
+
+        echo -n "$GPG_KEY_PEM" | gpg --batch --import
+        echo -e "5\ny\n" | gpg --command-fd 0 --edit-key "$GPG_KEY_ID" trust
```
Comment (hluotola-paf) [inline github/configure-git/action.yml]: What is this?

File: github/configure-git/action.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
+name: configure-git
+description: Configures git and gh
+
+inputs:
+  user_name:
```
Comment (hluotola-paf) [inline github/configure-git/action.yml:5]: Cosmetic: actions/checkout, actions/setup-python, actions/setup-node, ... all use `-` instead of `_`.

Comment (jpalomaki) [reply github/configure-git/action.yml:5]: Good to be consistent, will change to dash

File: github/configure-git/action.yml (hunk 1 above)
Comment (jpalomaki) [reply github/configure-git/action.yml]: Sets trust to ultimate for the imported key

**Review by hluotola-paf — APPROVED** [review]

**Review by lghakamo-paf — APPROVED** [review]
LGTM

**Pull request merged**